src/query/shorthand.ts

```diff
@@ -189,8 +189,8 @@ export function fieldDef(encQ: EncodingQuery,
   }
 
   // Scale
-  // TODO: axis, legend
-  for (const nestedPropParent of [Property.AXIS, Property.SCALE, Property.SORT]) {
+  // TODO: legend
+  for (const nestedPropParent of [Property.AXIS, Property.LEGEND, Property.SCALE, Property.SORT]) {
```

Comment:
SCALE, SORT, AXIS, LEGEND  (scale and sort are more important thus should come first)